json_object.c

```diff
@@ -591,7 +591,14 @@ json_bool json_object_get_boolean(const struct json_object *jso)
 	case json_type_boolean:
 		return jso->o.c_boolean;
 	case json_type_int:
-		return (jso->o.c_int64 != 0);
+		switch(jso->o.c_int.cint_type) {
+		case json_object_int_type_int64:
+			return (jso->o.c_int.cint.c_int64 != 0);
+		case json_object_int_type_uint64:
+			return (jso->o.c_int.cint.c_uint64 != 0);
+		default:
```

Comment:
if we're going to have explicit default/else clauses for cint_type checks so catch unexpected cint_type values then they should lead to assert() calls, not an arbitrary value return. 
However, it'd also be ok (and avoid dead code) to have it be simply e.g.:

    switch (jso->o.c_int.cint_type) {
    case json_object_int_type_int64:
        return (jso->o.c_int.cint.c_int64 != 0);
    default: // i.e. json_object_int_type_uint64
        return (jso->o.c_int.cint.c_uint64 != 0);

Reply:
I get it

Reply:
Though you adjusted this location, there are several other places father below that could use similar treatment.

Reply:
ok